app/src/main/java/net/squanchy/notification/NotificationCreator.kt

```diff
@@ -0,0 +1,203 @@
+        return event.track
+                .map { (_, _, accentColor) -> Color.parseColor(accentColor.or(ARGB_TRANSPARENT)) }
+                .or(Color.TRANSPARENT)
+    }
+
+    private fun createPendingIntentForMultipleEvents(): PendingIntent {
+        val taskBuilder = createBaseTaskStackBuilder()
+        return taskBuilder.getPendingIntent(0, PendingIntent.FLAG_CANCEL_CURRENT)
+    }
+
+    private fun createBaseTaskStackBuilder(): TaskStackBuilder {
+        val homescreenIntent = Intent(context, HomeActivity::class.java)
+        homescreenIntent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
+        return TaskStackBuilder.create(context)
+                .addNextIntent(homescreenIntent)
+    }
+
+    private fun createBigTextRichNotification(notificationBuilder: NotificationCompat.Builder, event: Event): NotificationCompat.BigTextStyle {
+        val bigTextBuilder = StringBuilder()
+                .append(getSpeakerNamesFrom(event.speakers))
+
+        val placeName = getPlaceName(event)
+        if (!placeName.isNullOrEmpty()) {
+            bigTextBuilder
+                    .append('\n')
+                    .append(context.getString(R.string.event_notification_starting_in, placeName))
+        }
+
+        return NotificationCompat.BigTextStyle(notificationBuilder)
+                .setBigContentTitle(event.title)
+                .bigText(bigTextBuilder.toString())
+    }
+
+    private fun getSpeakerNamesFrom(speakers: List<Speaker>): String {
+        val speakerNames = TextUtils.join(", ", map<Speaker, String>(speakers, { it.name }))
+        return context.getString(R.string.event_notification_starting_by, speakerNames)
+    }
+
+    private fun createInboxStyleRichNotification(notificationBuilder: NotificationCompat.Builder,
+                                                 events: List<Event>): NotificationCompat.InboxStyle {
+        val bigContentTitle = createSummaryTitle(events.size)
+        val richNotification = NotificationCompat.InboxStyle(notificationBuilder)
+                .setBigContentTitle(bigContentTitle)
+
+        for (event in events) {
+            if (event.place.isPresent) {
+                richNotification.addLine(
+                        context.getString(
+                                R.string.room_event_notification,
+                                event.place.get().name,
+                                event.title
+                        )
+                )
+            } else {
+                richNotification.addLine(event.title)
+            }
+        }
+
+        return richNotification
+    }
+
+    private fun createSummaryTitle(talksCount: Int): String {
+        val quantityString = context.resources
+                .getQuantityString(R.plurals.event_notification_count_starting, talksCount)
+        return String.format(quantityString, talksCount)
+    }
+
+}
+
+private val GROUP_KEY_NOTIFY_SESSION = "group_key_notify_session"
```

Comment:
I would put these (here and below) in a companion object for `NotificationCreator` instead, for encapsulation.